ingestion_tools/scripts/data_validation/fixtures/data.py

```diff
@@ -18,24 +23,98 @@
 # ==================================================================================================
 
 
-def get_header(mrcfile: str) -> MrcInterpreter:
-    """Get the mrc file header for a tilt series without downloading the entire file."""
-    fs = FileSystemApi.get_fs_api(mode="s3", force_overwrite=False)
-    with fs.open(mrcfile, "rb") as f:
-        header = MrcInterpreter(iostream=f, permissive=True, header_only=True)
-    return header
+def get_header(mrcfile: str, fs: FileSystemApi) -> MrcInterpreter:
+    """Get the mrc file headers for a list of mrc files."""
+    try:
+        with fs.open(mrcfile, "rb") as f:
+            header = MrcInterpreter(iostream=f, permissive=True, header_only=True)
+        return header
+    except Exception as e:
+        pytest.fail(f"Failed to get header for {mrcfile}: {e}")
+
+
+def get_zarr_headers(zarrfile: str, fs: FileSystemApi) -> Dict[str, Dict]:
+    """Get the zattrs and zarray data for a zarr volume file."""
+    expected_subfolders = {f"{zarrfile}/{i}" for i in range(3)}.union({f"{zarrfile}/.zattrs", f"{zarrfile}/.zgroup"})
+    actual_subfolders = {"s3://" + folder for folder in fs.glob(zarrfile + "/*")}
+    if expected_subfolders != actual_subfolders:
+        pytest.fail(f"Expected zarr subfolders: {expected_subfolders}, Actual zarr subfolders: {actual_subfolders}")
+
+    fsstore = zarr.storage.FSStore(url=zarrfile, mode="r", fs=fs.s3fs, dimension_separator="/")
+    fsstore_subfolders = set(fsstore.listdir())
+    expected_fsstore_subfolders = {str(i) for i in range(3)}.union({".zattrs", ".zgroup"})
+    if expected_fsstore_subfolders != fsstore_subfolders:
+        pytest.fail(f"Expected zarr subfolders: {expected_subfolders}, Actual zarr subfolders: {fsstore_subfolders}")
+
+    loc = ZarrLocation(fsstore)
+    zarrays = {}
+    for i in range(3):
+        zarrays[i] = json.loads(fsstore[str(i) + "/.zarray"].decode())
+    return {"zattrs": loc.root_attrs, "zarrays": zarrays}
 
 
-def get_headers(mrcfiles: List[str]) -> Dict[str, MrcInterpreter]:
-    """Get the mrc file headers for a list of mrc files."""
-    headers = {}
-    for file in mrcfiles:
-        try:
-            headers[file] = get_header(file)
-        except Exception as _:
-            pytest.fail(f"Failed to get header for {file}")
+# ==================================================================================================
+# Dataset fixtures
+# ==================================================================================================
 
-    return headers
```

Comment:
no need to retrieve multiple headers, since we never have multiple mrc files for a given entity